src/js/application.js

```diff
@@ -0,0 +1,288 @@
+import { selectors } from './constants/selectors';
+import { classes } from './constants/classes';
+import { Location } from './location';
+import { Weather } from './weather';
+import { weatherIcons } from './weather-icons';
+import { translation } from './translation';
+
+export class Application {
+    constructor() {
+        this.degreeUnit = localStorage.getItem('degreeUnit') || 'celsius';
+        this.language = 'en';
+
+        this.currentLocation;
+        this.currentWeather;
+    }
+
+    init() { 
+        this.initActiveDegreeButton();
+        const SEARCH_LOCATION_ELEMENT = document.querySelector(selectors.SEARCH_FORM);
+        SEARCH_LOCATION_ELEMENT.addEventListener('submit', this.onLocationChange.bind(this));  
+
+        const LANGUAGE_BUTTON_ELEMENT = document.querySelector(selectors.LANGUAGE_BTN);
+        LANGUAGE_BUTTON_ELEMENT.addEventListener('change', this.changeLanguage.bind(this)); 
+
+        const BUTTONS_CONTAINER = document.querySelector(selectors.BUTTONS_CONTAINER);
+        BUTTONS_CONTAINER.addEventListener('click', this.determineClickedButton.bind(this));
+        
+        this.changeBackground();
+        const savedLanguage = localStorage.getItem('language');
+        if (savedLanguage && savedLanguage != this.language) {
+            this.setLanguage(savedLanguage, false);
+        }
+        this.loadLocationByIP();
+    }
+
+    showErrorMessage() {
+        const ERROR_MESSAGE_ELEMENT = document.querySelector(selectors.ERROR_MESSAGE);
+        ERROR_MESSAGE_ELEMENT.style.display = 'block';
+    
+        setTimeout(() => {
+            ERROR_MESSAGE_ELEMENT.style.display = 'none';
+        }, 3000);    
+    }
+
+    async onLocationChange(e) {    
+        e.preventDefault();
+
+        const SEARCH_INPUT_ELEMENT = document.querySelector(selectors.SEARCH_INPUT_FIELD);
+        const location = await Location.getLocationByName(SEARCH_INPUT_ELEMENT.value, this.language);
+
+        if (location) {
+            await location.loadLocationTimezone();
```

Comment:
let's have instead 2 classes: location and current time/date/timezone

Reply:
![image](https://user-images.githubusercontent.com/25482018/115145877-9c9e7d80-a05c-11eb-9811-8cab6e709165.png)

Reply:
fixed